vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/main/java/com/vaadin/flow/component/spreadsheet/Spreadsheet.java

```diff
@@ -4804,22 +4819,24 @@ private boolean isCellVisible(int row, int col) {
     }
 
     private void registerPopupButton(PopupButton button) {
-        if (popupButtonsEnabled) {
-            attachedPopupButtons.add(button);
-            registerCustomComponent(button);
+        attachedPopupButtons.add(button);
+        registerCustomComponent(button);
+        if (!getElement().equals(button.getElement().getParent())) {
+            getElement().appendVirtualChild(button.getElement());
         }
```

Comment:
As mentioned by @tomivirkki on our call, I changed the approach of how the `PopupButton` is sent to the client to add as a virtual child of Spreadhsset.